lib/code_corps/helpers/query.ex

```diff
@@ -2,7 +2,12 @@ defmodule CodeCorps.Helpers.Query do
   import CodeCorps.Helpers.String, only: [coalesce_id_string: 1]
   import Ecto.Query, only: [where: 3, limit: 2, order_by: 2]
 
-  def id_filter(query, id_list) do
+  @spec id_filter(Ecto.Queryable.t, map | String.t) :: Ecto.Queryable.t
+  def id_filter(query, %{"filter" => %{"id" => id_csv}}) do
+    query |> id_filter(id_csv)
+  end
+  def id_filter(query, %{}), do: query
+  def id_filter(query, id_list) when is_binary(id_list) do
```

Comment:
This whole thing has potential to be rewritten in a recursive fashion, where we accept a map, then pass the queryable through a pipe of clauses, each supporting a single key, applying a filter from that key, then calling the next clause with a map where that key was removed. We could even split our queries apport into context-specific modules, for example, `Common.Query`, `Comment.Query`, etc.